### PR TITLE
[QA checks] tolerate connector without dockerfile

### PR DESCRIPTION
**Pull request by alafanechere**
# What

Now that it's OK that a connector does not have a Dockerfile our QA checks should tolerate it.

# How
- Expose a `has_dockerfile` property on the `Connector` utility class
- Do not run dockerfile related checks when a Dockerfile does not exist

### CODE REVIEW

**Comment by alafanechere** [comment]
Current dependencies on/for this PR:
* master
  * **PR #30303** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/30303" target="_blank">Graphite</a> 
    * **PR #30456** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/30456" target="_blank">Graphite</a> 
      * **PR #30467** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/30467" target="_blank">Graphite</a>  👈
      * **PR #30458** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/30458" target="_blank">Graphite</a> 
    * **PR #30449** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/30449" target="_blank">Graphite</a> 
      * **PR #30450** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/30450" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/30467?utm_source=stack-comment).

File: airbyte-ci/connectors/connector_ops/connector_ops/qa_checks.py [hunk 1]
```diff
@@ -250,8 +250,13 @@ def check_metadata_version_matches_dockerfile_label(connector: Connector) -> boo
     # https://github.com/airbytehq/airbyte/issues/21606
     check_connector_https_url_only,
     check_connector_has_no_critical_vulnerabilities,
-    check_metadata_version_matches_dockerfile_label,
-]
+)
+
+
+def get_qa_checks_to_run(connector: Connector) -> Tuple[Callable]:
```
Comment (bnchrch) [inline airbyte-ci/connectors/connector_ops/connector_ops/qa_checks.py:256]: 👍

**Review by bnchrch — APPROVED** [review]

File: airbyte-ci/connectors/connector_ops/connector_ops/utils.py [hunk 2]
```diff
@@ -253,22 +257,20 @@ def language(self) -> ConnectorLanguage:
...
-                    return ConnectorLanguage.JAVA
-        except FileNotFoundError:
-            pass
+        if Path(self.code_directory / "build.gradle").is_file():
+            return ConnectorLanguage.JAVA
         return None
 
     @property
-    def version(self) -> str:
+    def version(self) -> Optional[str]:
```
Comment (erohmensing) [inline airbyte-ci/connectors/connector_ops/connector_ops/utils.py:265]: why is this one optional? I get becuase `self.version_in_dockerfile_label` could now be `None` - but shouldn't we error out if we can't retrieve a version for the connector?

**Comment by airbyte-oss-build-runner** [comment]
##  source-faker test report (commit [`d8731ed979`](https://github.com/airbytehq/airbyte/pull/30467/commits/d8731ed979a802972bc03969c63dffd27a951bd4)) - ✅

⏲️  Total pipeline duration: 01mn58s 

| Step                                                         | Result   |
|:-------------------------------------------------------------|:---------|
| Connector package install                                    | ✅        |
| Build source-faker docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                   | ✅        |
| Acceptance tests                                             | ✅        |
| Code format checks                                           | ✅        |
| Validate metadata for source-faker                           | ✅        |
| Connector version semver check                               | ✅        |
| QA checks                                                    | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/augustin_09-15-make_qa_checks_tolerate_connector_without_dockerfile/1696946203/d8731ed979a802972bc03969c63dffd27a951bd4/source-faker/5.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:d8731ed979a802972bc03969c63dffd27a951bd4)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-faker test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-duckdb test report (commit [`d8731ed979`](https://github.com/airbytehq/airbyte/pull/30467/commits/d8731ed979a802972bc03969c63dffd27a951bd4)) - ✅

⏲️  Total pipeline duration: 02mn33s 

| Step                                                               | Result   |
|:-------------------------------------------------------------------|:---------|
| Connector package install                                          | ✅        |
| Build destination-duckdb docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                         | ✅        |
| Integration tests                                                  | ✅        |
| Acceptance tests                                                   | ✅        |
| Code format checks                                                 | ✅        |
| Validate metadata for destination-duckdb                           | ✅        |
| Connector version semver check                                     | ✅        |
| QA checks                                                          | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/augustin_09-15-make_qa_checks_tolerate_connector_without_dockerfile/1696946203/d8731ed979a802972bc03969c63dffd27a951bd4/destination-duckdb/0.2.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:d8731ed979a802972bc03969c63dffd27a951bd4)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-duckdb test
```

**Pull request merged**